File: 폰켓몬.py
class Solution {
    public int solution(int[] nums) {        
        int[] checks = new int[200000];
        int n = nums.length / 2;
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            checks[nums[i]-1] = 1; 
        }
        
        for (int j = 0; j < checks.length; j++) {
                if (checks[j] == 1) 
                    sum++;
            }
        
        if (sum >= n) {
            return n;
        } else {
            return sum;
        }
    }
}

File: 폰켓몬.cpp
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> nums) {
	int cnt = 1;
	int size = nums.size() /2;

	sort(nums.begin(), nums.end());

	for (int i = 1; i < nums.size(); i++) {
		if (nums[i-1] != nums[i]){
			cnt++;
		}		
	}

    if (cnt > size) {
    	return size;
    } else if (cnt == 0) {
    	return 1;
    } else {
    	return cnt;
    }
	
}